vimiv/imutils/exif.py

```diff
@@ -11,138 +11,267 @@
 """
 
 import contextlib
+from typing import Dict
 
 from vimiv.utils import log, lazy
 from vimiv import api
 
+pyexiv2 = lazy.import_module("pyexiv2", optional=True)
 piexif = lazy.import_module("piexif", optional=True)
-
 _logger = log.module_logger(__name__)
 
 
-def check_piexif(return_value=None):
-    """Decorator for functions that require the optional piexif module.
+def check_exif_dependancy(return_value=None, check_piexif=True):
+    """Decorator for ExifHandler which require the optional py3exiv2 module.
 
-    If piexif is not available, return_value is returned and a debug log message is
-    logged. It it is available, the function is called as usual.
+    If py3exiv2 is available the class is left as it is. If py3exiv2 is not available
+    but the depreciated piexif module is, a depreciation warning is given to the user
+    and a ExifHandler class supporting piexif is returned. If none of the two modules
+    is available, a dummy class is returned and a debug log is logged.
 
     Args:
-        return_value: Value to return if piexif is not available.
+        return_value: Value to return if neither py3exiv2 nor piexif is available.
+        check_piexif: Check piexif dependency too in case py3exiv2 is not available.
```

Comment:
Is there any reason to have `check_piexif` as a decorator argument? For testing?
Otherwise we can just remove the piexif part as soon as piexif support is removed, i.e. before `v.1.0.0`.

Reply:
I have added this only for the tests in `tests/unit/imutils/test_exif`. We can also get rid of this option and design the tests differently, no problem.

Reply:
I think we can get around both of these arguments by making smart use of pytest fixtures. I would assume that we currently:
- always want to check for piexif
- always want to return one of the three implementations of the `ExifHandler` class, never `None` or another funky return value

Reply:
Improved in bc5fe863a3641910b8bc3793b2a60e5fe2c8289c